### PR TITLE
Set terminal mode appropriately on CI

### LINKED ISSUE #2070: Improve the UX of provision/deploy in CI/CD
Today, we render the progress spinners in CI/CD, which looks a bit rough. It'd probably even be nicer to just see "starting" "in progress" and "done" markers for each step instead.

image

### PR DESCRIPTION
**Pull request by weikanglim**
When the process environment is determined to be on CI, disable interactive updates for our interactive spinner. Allow for a user override if this is undesirable. 

Fixes #2070

### CODE REVIEW

**Comment by weikanglim** [comment]
Will run a GitHub actions using these bits to see the final result before posting back.

**Review by ellismg — APPROVED** [review]
LGTM

File: cli/azd/pkg/input/console.go [hunk 1]
```diff
@@ -10,9 +10,11 @@ import (
 	"io"
 	"log"
 	"os"
+	"strconv"
 	"time"
 
 	"github.com/AlecAivazis/survey/v2"
+	"github.com/azure/azure-dev/cli/azd/internal/telemetry/resource"
```
Comment (vhvb1989) [inline cli/azd/pkg/input/console.go:17]: nit. Should we move `resource` outside of `telemetry` ?  Just for code hygiene, if `input` is using `resource`, it should be something that is consumed by telemetry and input, and not from telemetry...

File: cli/azd/internal/telemetry/resource/ci.go [hunk 2]
```diff
@@ -80,3 +80,8 @@ func execEnvForCi() string {
 
 	return ""
 }
+
+// IsRunningOnCI returns true if the current process is running on a CI/CD provider.
+func IsRunningOnCI() bool {
+	return execEnvForCi() != ""
+}
```
Comment (vhvb1989) [inline cli/azd/internal/telemetry/resource/ci.go:87]: I'd like if we could promote the `IsRunningOnCI` to a new package outside of telemetry.
Looking at `execEnvForCi`, is there anything it requires from telemetry which justifies having it as part of it?

**Review by vhvb1989 — APPROVED** [review]
Consider re-arranging the packages

File: cli/azd/pkg/input/console.go (hunk 1 above)
Comment (weikanglim) [reply cli/azd/pkg/input/console.go:18]: I do agree with you, but we should wait until we have enough other use cases to form a package. `resource` is fairly separated out from most of `telemetry`. The specific file we're linking to only references another package `fields` which are just string-constants. From an architecture point of view, I don't think we'll run into troubles here. It's purely hygiene based on package name + clean separation, which from a pragmatic standpoint isn't always necessary. 

I'll definitely keep in mind of this in the future.

File: cli/azd/internal/telemetry/resource/ci.go (hunk 2 above)
Comment (weikanglim) [reply cli/azd/internal/telemetry/resource/ci.go:87]: Explained in the other thread. It's fairly decoupled right now.

**Comment by weikanglim** [comment]
Looks pretty!
image

image

**Pull request merged**